DuggaSys/gitfetchService.php

```diff
@@ -142,38 +164,47 @@ function bfs($url, $cid, $opt)
         // Starts a stream with the required headers
         $context = stream_context_create($opts);
         // Fetches the data with the stream included
-        $data = @file_get_contents($currentUrl, true, $context);
+        $data = @file_get_contents($currentUrl, true, $context); 
         if ($data) {
             // Decodes the fetched data into JSON
             $json = json_decode($data, true);
+            if($opt == "GETCOMMIT"){
+                return $json['sha'];
+            }
             // Loops through each item fetched in the JSON data
             if ($json) {
                 foreach ($json as $item) {
                     // Checks if the fetched item is of type 'file'
                     if ($item['type'] == 'file') {
                         //If an index file has been found, check against the content of the index file
-                        if($filesToIgnore){
-                            //If file is not part of files to ignore, resume (Index file)
-                            if(!in_array($item['name'], $filesToIgnore)){
-                                if($opt == "REFRESH") {
-                                    insertToMetaData($cid, $item);
+                        if ($item['name'] != ".gitignore") {
+                            if ($filesToIgnore) {
+                                // If the file is not part of files to ignore, resume (Index file)
+                                if (!in_array($item['name'], $filesToIgnore) && !in_array($item['name'], $filesVisited)) {
+                                    if ($opt == "REFRESH") {
+                                        insertToMetaData($cid, $item);
+                                        array_push($filesVisited,$item['name']);
+                                    } else if ($opt == "DOWNLOAD") {
+                                        insertToFileLink($cid, $item);
+                                        insertToMetaData($cid, $item);
+                                        downloadToWebserver($cid, $item);
+                                        array_push($filesVisited,$item['name']);
```

Comment:
Somewhere here it should log wether or not the file was downloaded correctly, so that its possible to know if the code failed, or if the server was rate limited by github. 

Perhaps add some error handling for the `file_get_contents()` function used in `downloadToWebServer()` and return an error if it failed. then log that somewhere using one of the already existing logging functions in LenaSYS. Here is a relevant thread about [handling errors from `file_get_contents()`](https://stackoverflow.com/questions/272361/how-can-i-handle-the-warning-of-file-get-contents-function-in-php).